### DISCUSSION TITLE
Load csv columns at end of file with no values

### DISCUSSION THREAD

**Discussion by fsuoj** [PHPOffice/PhpSpreadsheet#2839] [Q&A]
I have a csv file that has columns at the end with no data.  The IOFactory load will not create the array columns for these final columns.  If I add values to the empty columns then it generates array entries.  Here is the example:

$inputFileType = IOFactory::indentify($inputFileName);
$reader = IOFactory::createReader($inputFileType);
$spreadSheet = $reader->load($inputFileName);
var_dump($spreadSheet[1]);
die();

CSV Data - 
John,Doe,,
Mary,Jane,,

If you var_dump($spreadSheet[1]); it will only return the first two keys and not the final two keys.
If I change the data to be the following:
John,Doe,A,1
Mary,Jane,A,2

Then you will see the full array of data in the var_dump results.  

Is there a setting I am missing in the IOFactory methods to include all columns?

Owen

**Comment by oleibman** [comment]
From one Owen to another, no, you are not missing any settings. You could certainly open an issue to request such a feature, or, even better, a PR to implement it. It does not look like it would be difficult to add.

**Reply by fsuoj** [reply]
Thank you Owen!  Our name seems to becoming a more popular name, glad we could set the standard.  

I need the ability to refer to all columns even if the header row does not exist.  It seems weird that they would exclude the final columns from the array.  I have seen where people have asked to exclude these but when you are mapping data and you are expecting $spreadsheet[1]["C"] and the key does not exist, this is a problem.  I will look at their code and see if I can come up with a solution.

**Comment by oleibman** [comment]
This was implemented by PR #2842, which has been part of the official release since 1.24.
